src/url-helper.ts

```diff
@@ -21,26 +21,56 @@ export function getFetchUrl(settings: IGitSourceSettings): string {
 }
 
 export function getServerUrl(url?: string): URL {
-  let urlValue =
-    url && url.trim().length > 0
-      ? url
-      : process.env['GITHUB_SERVER_URL'] || 'https://github.com'
-  return new URL(urlValue)
+  let resolvedUrl = process.env['GITHUB_SERVER_URL'] || 'https://github.com'
+  if (hasContent(url, false)) {
+    resolvedUrl = url!
+  }
+
+  return new URL(resolvedUrl)
 }
 
 export function getServerApiUrl(url?: string): string {
-  let apiUrl = 'https://api.github.com'
+  if (hasContent(url, false)) {
```

Comment:
Style nit: `hasContent` here is a bit confusing and I don't think it adds much value.

The checks for `hasContent(url, false)` (where  `allowPureWhitespace` is false) are relatively simple and could be inlined to be a bit more readable.

`if (hasContent(url, false)) {` is equivalent to `if (url) {`, the latter does not require the reader to go figure out what's being set to `false` and knowing how `hasContent` works.

If we removed these cases, we could simplify `hasContent` to always trim and likely inline that usage as well.

```diff
diff --git a/src/url-helper.ts b/src/url-helper.ts
index 5c4fcbd..27969ca 100644
--- a/src/url-helper.ts
+++ b/src/url-helper.ts
@@ -22,15 +22,15 @@ export function getFetchUrl(settings: IGitSourceSettings): string {
 
 export function getServerUrl(url?: string): URL {
   let resolvedUrl = process.env['GITHUB_SERVER_URL'] || 'https://github.com'
-  if (hasContent(url, false)) {
-    resolvedUrl = url!
+  if (url) {
+    resolvedUrl = url
   }
 
   return new URL(resolvedUrl)
 }
 
 export function getServerApiUrl(url?: string): string {
-  if (hasContent(url, false)) {
+  if (url) {
     let serverUrl = getServerUrl(url)
     if (isGhes(url)) {
       serverUrl.pathname = 'api/v3'
@@ -58,19 +58,8 @@ export function isGhes(url?: string): boolean {
 }
 
 function pruneSuffix(text: string, suffix: string) {
-  if (hasContent(suffix, true) && text?.endsWith(suffix)) {
+  if (suffix.trim() && text?.endsWith(suffix)) {
     return text.substring(0, text.length - suffix.length)
   }
   return text
 }
-
-function hasContent(
-  text: string | undefined,
-  allowPureWhitespace: boolean
-): boolean {
-  let refinedText = text ?? ''
-  if (!allowPureWhitespace) {
-    refinedText = refinedText.trim()
-  }
-  return refinedText.length > 0
-}

```

Reply:
| `str` | `if(str)` | `hasContent(str, true)` | `hasContent(str, true)` |
|-:|-:|-:|-:|
| `null` | `false` | `false` | `false` |
| `undefined` | `false` | `false` | `false` |
| `''` | `false` | `false` | `false` |
| `'abc'` | `true` | `true` | `true` |
| `' '` | `true` | `true` | `false` |

Reply:
It's the other way 'round.  A string containing pure whitespace is "truthy," so the predominant pattern would be `if(url?.trim()) { ... }`

I found `hasContent(url, bool)` to be more readable.

Reply:
fixed with an enum